### ISSUE TITLE
Revamp firewall management

### ISSUE THREAD

**Issue by Salamandar** [YunoHost/issues#2497]
- [ ] Stop differentiating ipv4/ipv6 (ipv6 should be first citizen, and IP protocols should be handled completely or not at all, not partially)
- [ ] Handle ports per-app (for now disallow == delete, we should have a separate operation)
- [ ] Fix uPnP
  - [ ] localport=1
  - [ ] ~don't try ports under 1024 (or at least don't fail on them)~
  - [ ] maybe detect automatically the allowed port range instead of failing on the first?
  - [ ] on `admin/#/tools/firewall`, set the uPnP as checkable (like ipv4/ipv6) (and rename the column like "publicly visible")
- [ ] Use nftables instead of iptables
  - [ ] maybe a first stupid simple step with the nftables-iptables cmdline ?
  - [ ] via regen-conf
  - [ ] fail2ban ?
  - [ ] drop the yunohost-firewall service in favor of the nftables one ?

**Comment by alexAubin** [comment]
> don't try ports under 1024 (or at least don't fail on them)

The point of the UPnP integration is that people won't need to manually configure port 22, 25, 80, 443 and 993(? never remember the IMAP/SMTP one..). There's little interest in supporting it or even having flagged as "success" if neither of these worked

**Comment by alexAubin** [comment]
>drop the yunohost-firewall service in favor of the nftables one ?

dunno what you mean but yeah there's `ufw` and `firewalld` which may be better to integrate rather than having our own custom stuff ... I don't know enough about these to know the pro and cons versus having our own custom stuff though

**Comment by Salamandar** [comment]
> The point of the UPnP integration is that people won't need to manually configure port 22, 25, 80, 443 and 993

Yeah i get what you mean… But IMHO relying on the behaviour of routers that allow uPnP to configure ports lower than 1024 sounds fishy to me. That looks like a *very dangerous* behaviour and I would be very surprised if that was broadly supported.

If only there was a lib like `dns lexicon` for routers config… O:)

Anyways, i stroked the line from the checklist.

> dunno what you mean but yeah there's ufw and firewalld which may be better to integrate rather than having our own custom stuff

Well nftables comes with a systemd service that basically does the same stuff than yunohost-firewall (tl;dr oneshot service that runs when the network goes up and applies the config). `ufw` is exactly the same stuff too, the difference being that nftables is the official debian way to go. Having a yunohost-firewall service just seems redundant to me.

**Comment by Axolotle** [comment]
@Salamandar can we close this?

**Comment by alexAubin** [comment]
Assuming it's done
